src/ng/compile.js

```diff
@@ -1710,13 +1710,20 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
 
         nodeName = nodeName_(this.$$element);
 
-        if ((nodeName === 'a' && (key === 'href' || key === 'xlinkHref')) ||
-            (nodeName === 'img' && key === 'src') ||
-            (nodeName === 'image' && key === 'xlinkHref')) {
-          // sanitize a[href] and img[src] values
-          this[key] = value = $$sanitizeUri(value, nodeName === 'img' || nodeName === 'image');
```

Comment:
So, doesn't this leave people using `ng-href/src` less vulnerable (unless they use interpolation)?
(Not common to use `ng-href/src` without interpollation, but possible.)

Reply:
I am not sure what you mean here. Do you mean "more" vulnerable?
It is true that if you were programmatically setting the attribute via `$set` then it is no longer checked, but this is quite a common paradigm, where one has to be responsible for coding against the lower level API.

Reply:
Yeah, I meant "more" (everybody knows "less" is "more" 😛)
The thing is that some of the built-in directives (e.g. `ngHref``/`ngSrc`) use `$set` under the hood.

Reply:
I am pretty certain that `ngHref` and `ngSrc` are not affected. They only call `$set` with a value that is passed from `$observe`, which has already interpolated the value, and so has passed the value through SCE.